### DISCUSSION TITLE
[Application Development] How can a smart contract programmatically bridge ETH from an Optimism L2 to Ethereum L1 using the Optimism bridge?

### DISCUSSION THREAD

**Discussion by Zodomo** [ethereum-optimism/developers#427] [Application Development] [answered]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I want to programmatically bridge ETH from an Optimism L2 to Ethereum L1 via a smart contract. Based on what superchain bridge interfaces prompt wallets with, they're calling the `bridgeETHTo` function on the L2StandardBridge contract. However, after one of these transactions is included in an L1 state root, you need to prove that it is by calling `proveWithdrawalTransaction` on the OptimismPortal contract on Ethereum L1. How can I produce the inclusion proofs for `proveWithdrawalTransaction` for a transaction I did not submit and did not prepare via the Optimism SDK (as the smart contract is what initiated it)?

### Feedback

I cannot find any documentation for how to interact with the superchain bridge via smart contracts. Only offchain via your SDK.

**Comment by hash1go** [comment, marked as answer]
Hello!

If not using the Optimism SDK, you will need to manually create a Merkle proof to prove that the withdrawal transaction is included in the L2 state. This process will require off-chain resources.
Please refer to [the relevant specifications](https://specs.optimism.io/protocol/withdrawals.html) for more details.

**Reply by sbvegan** [reply]
You can also utilize viem: https://viem.sh/op-stack/guides/withdrawals
